src/test/store/receive-profile.test.js

```diff
@@ -1360,6 +1367,188 @@ describe('actions/receive-profile', function() {
       expect(nodeData.selfTime).toBe(4);
     });
   });
+
+  describe('getProfilesFromRawUrl', function() {
+    function fetch200Response(profile: string) {
+      return {
+        ok: true,
+        status: 200,
+        headers: {
+          get: () => 'application/json',
+        },
+        json: () => Promise.resolve(JSON.parse(profile)),
+      };
+    }
+
+    async function setup(location: Object, requiredProfile: number = 1) {
+      const profile = _getSimpleProfile();
+      const geckoProfile = createGeckoProfile();
+
+      // Add mock fetch response for the required number of times.
+      // Usually it's 1 but it can be also 2 for `compare` dataSource.
+      for (let i = 0; i < requiredProfile; i++) {
+        window.fetch.mockResolvedValueOnce(
+          fetch200Response(serializeProfile(profile))
+        );
+      }
+
+      const geckoProfiler = {
+        getProfile: jest.fn().mockResolvedValue(geckoProfile),
+        getSymbolTable: jest
+          .fn()
+          .mockRejectedValue(new Error('No symbol tables available')),
+      };
+      window.geckoProfilerPromise = Promise.resolve(geckoProfiler);
+
+      simulateSymbolStoreHasNoCache();
+
+      // Silence the logs coming from the promise rejections above.
+      jest.spyOn(console, 'log').mockImplementation(() => {});
+
+      const store = blankStore();
+      await store.dispatch(getProfilesFromRawUrl(location));
+
+      // To find stupid mistakes more easily, check that we didn't get a fatal
+      // error here. If we got one, let's rethrow the error.
+      const view = getView(store.getState());
+      if (view.phase === 'FATAL_ERROR') {
+        throw view.error;
+      }
+
+      const waitUntilPhase = phase =>
+        waitUntilState(store, state => getView(state).phase === phase);
+
+      const waitUntilSymbolication = () =>
+        waitUntilState(
+          store,
+          state => ProfileViewSelectors.getSymbolicationStatus(state) === 'DONE'
+        );
+
+      return {
+        profile,
+        geckoProfile,
+        waitUntilPhase,
+        waitUntilSymbolication,
+        ...store,
+      };
+    }
+
+    beforeEach(function() {
+      // The stub makes it easy to return different values for different
+      // arguments. Here we define the default return value because there is no
+      // argument specified.
+      window.fetch = jest.fn();
```

Comment:
Can we use `jest.spyOn` here as well?

Reply:
Same for this as well. We can't use it because it's not implemented in `JSDOM` :(